### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by andresilva**
- Updates some dependencies
- Migrates to quickcheck 1.0
- Makes the fuzzing tests work again

Merge after #128.

### CODE REVIEW

File: fuzz/afl_targets/graph.rs [hunk 1]
```diff
@@ -12,11 +12,13 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#[macro_use]
+extern crate afl;
```
Comment (thiolliere) [inline fuzz/afl_targets/graph.rs:16]: I actually tend to prefer not importing macro and just calling them from their path, but I'm fine anyway,

File: src/bitfield.rs [hunk 2]
```diff
@@ -201,21 +207,36 @@ pub struct Bit1 {
...
 	}
 
 	#[test]
 	fn set_bit() {
 		fn prop(mut a: Bitfield, idx: usize) -> bool {
+			// let's bound the max index at 1000 times the number of bits in the
+			// bitfield. this is needed because otherwise when calling `set_bit`
+			// with a large value we might try to allocate more memory than we
+			// have.
+			let idx = idx.min(a.capacity() * 1000);
```
Comment (thiolliere) [inline src/bitfield.rs:238]: wouldn't it make more sense to have fixed limit like `2^20` or something?
Anyway it is fine.

**Review by thiolliere — APPROVED** [review]
looks good to me, I'm not familiar with the code base but seems legit.

File: fuzz/afl_targets/graph.rs (hunk 1 above)
Comment (andresilva) [reply fuzz/afl_targets/graph.rs:16]: Me too! But I needed to make this change to make it compile, `fuzz` is calling another macro inside it `__fuzz`.

File: src/bitfield.rs (hunk 2 above)
Comment (andresilva) [reply src/bitfield.rs:238]: Yeah good idea.

File: fuzz/afl_targets/graph.rs (hunk 1 above)
Comment (thiolliere) [reply fuzz/afl_targets/graph.rs:16]: ah I see

**Comment by andresilva** [comment]
Thanks for the review. Updated with your suggestion. It seems that quickcheck's search has gotten smarter in the meantime since some of the edge cases that I had to close now in `Arbitrary` weren't found before :P

File: README.md [hunk 3]
```diff
@@ -73,6 +73,36 @@ Most importantly this crate does not deal with authority set changes. It assumes
...
+```sh
+cargo install cargo-fuzz
+cargo install afl
+```
+
+#### libfuzzer
+
+```sh
+cargo fuzz run graph
+cargo fuzz run round
```
Comment (octol) [inline README.md:89]: This might require nightly rustc? Is that worth mentioning, or perhaps a note about it is expected to be obsolete soon anyway?

**Review by octol — APPROVED** [review]
Looks good!

File: README.md (hunk 3 above)
Comment (andresilva) [reply README.md:90]: Good point, I only use nightly so didn't notice it. Updated the readme.

**Review by rphmeier — APPROVED** [review]

**Pull request merged**